.github/workflows/install_and_test.yml

```diff
@@ -0,0 +1,241 @@
+    - main
+    tags: [ "v*" ]
+  schedule:
+    - cron: "0 5 * * *"
+    #this is 9PM PST
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4.1.1
+    - name: Setup python
+      uses: actions/setup-python@v5.0.0
+      with:
+        python-version: '3.9.16'
+    - name: install dependencies
+      run: |-
+        pip install -U pip setuptools pre-commit
+        # Install the hooks now so that they'll be cached
+        pre-commit install-hooks
+    - name: Check Code Style using pre-commit
+      run: |-
+        SKIP=eslint pre-commit run --show-diff-on-failure --all-files
+
+  python_lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4.1.1
+    - name: Setup python
+      uses: actions/setup-python@v5.0.0
+      with:
+        python-version: '3.9.16'
+    - name: setup
+      run: |-
+        pip install black==23.1.0 --progress-bar off
+        pip install "isort[pyproject]" numpy --progress-bar off
+        pip install mypy==0.991 types-mock types-Pillow types-tqdm types-PyYAML --progress-bar off
+        pip install -r habitat-lab/requirements.txt --progress-bar off
+    - name: run black
+      run: |-
+        black --version
+        black --exclude '/(\.eggs|\.git|\.hg|\.mypy_cache|\.nox|\.tox|\.venv|_build|buck-out|build|dist)|examples/tutorials/(notebooks|nb_python)' habitat-lab/. habitat-baselines/. examples/. test/. --diff
+        black --exclude '/(\.eggs|\.git|\.hg|\.mypy_cache|\.nox|\.tox|\.venv|_build|buck-out|build|dist)|examples/tutorials/(notebooks|nb_python)' habitat-lab/. habitat-baselines/. examples/. test/. --check
+    - name: run isort
+      run: |-
+        isort --version
+        isort habitat-lab/. habitat-baselines/. examples/. test/.  --diff
+        isort habitat-lab/. habitat-baselines/. examples/. test/.  --check-only
+    - name: run mypy
+      run: |-
+        mypy --version
+        mypy --exclude="^docs/|setup.py$"
+
+  install_and_test_lab_ubuntu:
+    runs-on: 4-core-ubuntu-gpu-t4
+    defaults:
+      run:
+        shell: bash -el {0}
+    steps:
+    - uses: actions/checkout@v4.1.1
+      with:
+        path: "./habitat-lab"
+    - uses: "./habitat-lab/.github/actions/install_ubuntu_deps"
+    - uses: "./habitat-lab/.github/actions/install_ubuntu_gpu_deps"
+    - name: Build, install habitat-sim
+      run: |-
+        #give cmake ownership to the runner for installation
+        sudo chown runner -R /opt/cmake312/
+        #activate conda env
+        export PATH=$HOME/miniconda/bin:/usr/local/cuda/bin:$PATH
+        conda activate habitat
+        #install habitat-sim
+        if [ ! -d ./habitat-sim ]
+        then
+          git clone https://github.com/facebookresearch/habitat-sim.git --recursive
+        fi
+        cd habitat-sim
+        pip install -r requirements.txt --progress-bar off
+        git submodule update --init --recursive --jobs 8
+        python -u setup.py install --headless --with-cuda --bullet
```

Comment:
Would it make sense to run tests using the latest successful sim conda build?

It would reduce build times and independent failures from sim main.

Reply:
Sure, that makes sense. I was thinking the opposite, that waiting for changes to hit nightly is a delay and possible pain point. For example, if I have a pair of sim-lab PRs that accomplish a refactor and I merge the sim PR, I need to wait 24 hours to merge the lab PR or live with breaking CI for that duration. :shrug: 

We can discuss and change this if we want. :+1: